manifests/init.pp

```diff
@@ -134,8 +135,89 @@
   Boolean                       $simp_dashboards         = false
 ) inherits ::simp_grafana::params {
 
-  $merged_cfg = deep_merge($::simp_grafana::params::cfg, $cfg)
-  $merged_ldap_cfg = deep_merge($::simp_grafana::params::ldap_cfg, $ldap_cfg)
+
+  $admin_pw = passgen('grafana')
+
+  $app_pki_dir             = '/etc/pki/simp_apps/grafana/x509'
+  $app_pki_key             = "${app_pki_dir}/private/${facts['fqdn']}.pem"
+  $app_pki_cert            = "${app_pki_dir}/public/${facts['fqdn']}.pub"
+
+  if ($ldap){
+    $base_dn = simplib::lookup('simp_options::ldap::base_dn', { 'default_value' => simplib::ldap::domain_to_dn() } )
```

Comment:
The reason why we did not make them class parameters is because if we don't have ldap configured they would throw an error, so they have to be encapsulated in the if statement. Check with Dylan if you still believe it's a problem.